src/core/net/ip4_types.hpp

```diff
@@ -0,0 +1,672 @@
+     * example, "127.0.0.1"
+     *
+     * @param[in]  aString        A pointer to the null-terminated string.
+     *
+     * @retval kErrorNone         Successfully parsed the IPv4 address string.
+     * @retval kErrorParse        Failed to parse the IPv4 address string.
+     *
+     */
+    Error FromString(const char *aString);
+
+    /**
+     * This method converts the IPv4 address to a string.
+     *
+     * The string format uses quad-dotted notation of four bytes in the address (e.g., "127.0.0.1").
+     *
+     * @returns An `InfoString` representing the IPv4 address.
+     *
+     */
+    InfoString ToString(void) const;
+} OT_TOOL_PACKED_END;
+
+class Cidr : public otIp4Cidr, public Unequatable<Cidr>, public Clearable<Address>
+{
+    friend class Address;
+
+public:
+    static constexpr uint16_t kCidrSuffixSize = 3; ///< Suffix to represent CIDR (/dd).
+
+    /**
+     * This type defines the fixed-length `String` object returned from `ToString()`.
+     *
+     */
+    typedef String<Address::kAddressStringSize + kCidrSuffixSize> InfoString;
+
+    /**
+     * This method converts the IPv4 CIDR to a string.
+     *
+     * The string format uses quad-dotted notation of four bytes in the address with the length of prefix (e.g.,
+     * "127.0.0.1/32").
+     *
+     * @returns An `InfoString` representing the IPv4 cidr.
+     *
+     */
+    InfoString ToString(void) const;
+
+    /**
+     * This method gets the prefix as a pointer to a byte array.
+     *
+     * @returns A pointer to a byte array containing the Prefix.
+     *
+     */
+    const uint8_t *GetBytes(void) const { return mAddress.mFields.m8; }
+
+    /**
+     * This method overloads operator `==` to evaluate whether or not two prefixes are equal.
+     *
+     * @param[in]  aOther  The other prefix to compare with.
+     *
+     * @retval TRUE   If the two prefixes are equal.
+     * @retval FALSE  If the two prefixes are not equal.
+     *
+     */
+    bool operator==(const Cidr &aOther) const;
+
+    /**
+     * This method sets the CIDR.
+     *
+     * @param[in] aAddress  A pointer to buffer containing the CIDR bytes. The length of aAddress should be 4 bytes.
+     * @param[in] aLength  The length of CIDR in bits.
+     *
+     */
+    void Set(const uint8_t *aAddress, uint8_t aLength);
+
+private:
+    /**
+     * This method returns the host mask (bitwise not of the subnet mask) of the CIDR.
+     *
+     * @returns A uint32 for the host mask, in network byte order.
+     *
+     */
+    uint32_t HostMask(void) const
+    {
+        // Note: Using LL suffix to make it a uint64 since /32 is a valid CIDR, and right shifting 32 bits is undefined
+        // for uint32.
+        return HostSwap32(0xffffffffLL >> mLength);
+    }
+
+    /**
+     * This method returns the subnet mask of the CIDR.
+     *
+     * @returns A uint32 for the subnet mask, in network byte order.
+     *
+     */
+    uint32_t SubnetMask(void) const { return ~HostMask(); }
```

Comment:
nit: no need to document private methods
```suggestion
    uint32_t SubnetMask(void) const { return ~HostMask(); }
```